crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -770,36 +770,23 @@ impl<'src> Parser<'src> {
     ///
     /// If the parser isn't positioned at a `type` token.
     ///
-    /// See: <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-python-grammar-type_stmt>
+    /// See: <https://docs.python.org/3/reference/simple_stmts.html#the-type-statement>
     fn parse_type_alias_statement(&mut self) -> ast::StmtTypeAlias {
         let start = self.node_start();
         self.bump(TokenKind::Type);
 
-        let (tok, tok_range) = self.next_token();
-        let name = if let Tok::Name { name } = tok {
-            Expr::Name(ast::ExprName {
-                id: name.to_string(),
-                ctx: ExprContext::Store,
-                range: tok_range,
-            })
-        } else {
-            // TODO(dhruvmanila): This recovery isn't possible currently because the soft keyword
-            // transformer will always convert the `type` token to a `Name` token if it's not
-            // followed by a `Name` token.
-            self.add_error(
-                ParseErrorType::OtherError(format!("expecting identifier, got {tok}")),
-                tok_range,
-            );
-            Expr::Name(ast::ExprName {
-                id: String::new(),
-                ctx: ExprContext::Invalid,
-                range: tok_range,
-            })
-        };
+        let mut name = Expr::Name(self.parse_name());
+        helpers::set_expr_ctx(&mut name, ExprContext::Store);
+
         let type_params = self.try_parse_type_params();
 
         self.expect(TokenKind::Equal);
 
+        // TODO(dhruvmanila): Disallow yield and starred expression
+        // test_err type_alias_invalid_value_expr
+        // type x = *y
+        // type x = yield y
+        // type x = x := 1
         let value = self.parse_conditional_expression_or_higher();
```

Comment:
Are there other places where we parse a "type" expression? E.g. when parsing type arguments, type annotations etc. If so, I think it would make sense to have a `parse_type` or `parse_type_expression` method.

Reply:
I think the main value is to have a place where we can reuse some of the expression validation logic and that I as a parse rule author don't always have to think about which expressions are valid (I can just call `parse_type_expression` and I know the parser will disallow expressions that aren't valid in a type context)

Reply:
Right, that makes sense but I think the `parse_conditional_expression_or_higher` function should, in the future, do these validations.